tst/EnergyPlus/unit/SurfaceGeometry.unit.cc

```diff
@@ -8860,6 +8860,167 @@ TEST_F(EnergyPlusFixture, SurfaceGeometry_ZoneAndSpaceAreas)
+        "	0.9,                     !- Thermal Absorptance              ",
+        "	0.8,                     !- Solar Absorptance                ",
+        "	0.8;                     !- Visible Absorptance              ",
+        "                                                                ",
+        "WindowMaterial:Glazing,                                                          ",
+        "	CLEAR 3MM,               !- Name                                              ",
+        "	SpectralAverage,         !- Optical Data Type                                 ",
+        "	,                        !- Window Glass Spectral Data Set Name               ",
+        "	0.003,                   !- Thickness {m}                                     ",
+        "	0.837,                   !- Solar Transmittance at Normal Incidence           ",
+        "	0.075,                   !- Front Side Solar Reflectance at Normal Incidence  ",
+        "	0.075,                   !- Back Side Solar Reflectance at Normal Incidence   ",
+        "	0.898,                   !- Visible Transmittance at Normal Incidence         ",
+        "	0.081,                   !- Front Side Visible Reflectance at Normal Incidence",
+        "	0.081,                   !- Back Side Visible Reflectance at Normal Incidence ",
+        "	0.0,                     !- Infrared Transmittance at Normal Incidence        ",
+        "	0.84,                    !- Front Side Infrared Hemispherical Emissivity      ",
+        "	0.84,                    !- Back Side Infrared Hemispherical Emissivity       ",
+        "	0.9;                     !- Conductivity {W/m-K}                              ",
+    });
+
+    // Prepare data for the test
+    ASSERT_TRUE(process_idf(idf_objects));
+    GetMaterialData(*state, ErrorsFound); // read material data
+    EXPECT_FALSE(ErrorsFound);
+    GetConstructData(*state, ErrorsFound); // read construction data
+    EXPECT_FALSE(ErrorsFound);
+    GetZoneData(*state, ErrorsFound); // read zone data
+    EXPECT_FALSE(ErrorsFound);
+    GetProjectControlData(*state, ErrorsFound); // read project control data
+    EXPECT_FALSE(ErrorsFound);
+    state->dataSurfaceGeometry->CosZoneRelNorth.allocate(1);
+    state->dataSurfaceGeometry->SinZoneRelNorth.allocate(1);
+    state->dataSurfaceGeometry->CosZoneRelNorth(1) = std::cos(-state->dataHeatBal->Zone(1).RelNorth * DataGlobalConstants::DegToRadians);
+    state->dataSurfaceGeometry->SinZoneRelNorth(1) = std::sin(-state->dataHeatBal->Zone(1).RelNorth * DataGlobalConstants::DegToRadians);
+    state->dataSurfaceGeometry->CosBldgRelNorth = 1.0;
+    state->dataSurfaceGeometry->SinBldgRelNorth = 0.0;
+    GetSurfaceData(*state, ErrorsFound); // setup zone geometry and get zone data
+    EXPECT_FALSE(ErrorsFound);           // expect no errors
+
+    EXPECT_NEAR(state->dataHeatBal->Zone(1).FloorArea, 1.0, 0.001);
+    EXPECT_NEAR(state->dataHeatBal->Zone(1).CalcFloorArea, 1.0, 0.001);
+}
+
```

Comment:
Unit test to show zone Floor Area are assigned correctly and is equal to the CalcFloorArea.

Reply:
So assuming we have two different variables holding floor area, then this unit test properly checks that.  It's OK for now.